src/lib/button/_button-theme.scss

```diff
@@ -54,20 +58,15 @@
   $foreground: map-get($theme, foreground);
 
   .mat-button, .mat-icon-button, .mat-raised-button, .mat-fab, .mat-mini-fab {
-    &.cdk-keyboard-focused {
```

Comment:
Removing this line means that the overlay *always* has the correct coloring applied. [`%mat-button-base`](https://github.com/willshowell/material2/blob/fix-buttons-hover/src/lib/button/_button-base.scss#L58-L62) takes care of showing/hiding it when the button has `.cdk-keyboard-focused`.